### ISSUE TITLE
The overhead cost so much time

### ISSUE THREAD

**Issue by satchelwu** [triton-inference-server/server#2137]
I used perf_client to test perfomance, but the overthead cost extremely too much time with perf_client running on both server host and other machine. the details is followed below.

using triton-inference-server v2.3.0dev
### model config
name: "test_model"
platform: "tensorrt_plan"
max_batch_size: 8
dynamic_batching {
  preferred_batch_size: [2, 4, 8]
  max_queue_delay_microseconds: 0
}
input [
  {
    name: "data"
    data_type: TYPE_FP32
    format: FORMAT_NHWC
    dims: [ 384, 640, 3 ]
  }
]
output[
  {
    name:"prob"
    data_type:TYPE_FP32
    dims:[6001,1,1]
  }
]

instance_group [
  {
    count:8
    gpus:1
    kind: KIND_GPU
  }
]

### server start config
--allow-http=true  --allow-grpc=true --grpc-infer-allocation-pool-size=100  --strict-readiness=false --log-verbose=3 --log-info=true --log-warning=true

### perf_client config
./perf_client  -m test_model --concurrency-range 1 -u 192.168.54.181:8000

### performance result from client side
*** Measurement Settings ***
  Batch size: 1
  Measurement window: 5000 msec
  Using synchronous calls for inference
  Stabilizing using average latency

Request concurrency: 1
  Client:
    Request count: 41
    Throughput: 8.2 infer/sec
    Avg latency: 120590 usec (standard deviation 20704 usec)
    p50 latency: 115669 usec
    p90 latency: 139971 usec
    p95 latency: 143088 usec
    p99 latency: 225839 usec
    Avg HTTP time: 118252 usec (send/recv 1311 usec + response wait 116941 usec)
  Server:
    Inference count: 51
    Execution count: 51
    Successful request count: 51
    Avg request latency: 517080879457 usec (overhead 517080863048 usec + queue 94 usec + compute input 4040 usec + compute infer 12199 usec + compute output 76 usec)

Inferences/Second vs. Client Average Batch Latency
Concurrency: 1, throughput: 8.2 infer/sec, latency 120590 usec

### performance result from server side
*** Measurement Settings ***
  Batch size: 1
  Measurement window: 5000 msec
  Using synchronous calls for inference
  Stabilizing using average latency

Request concurrency: 1
  Client:
    Request count: 392
    Throughput: 78.4 infer/sec
    Avg latency: 12744 usec (standard deviation 182 usec)
    p50 latency: 12711 usec
    p90 latency: 12813 usec
    p95 latency: 12940 usec
    p99 latency: 13082 usec
    Avg HTTP time: 12744 usec (send/recv 1949 usec + response wait 10795 usec)
  Server:
    Inference count: 470
    Execution count: 470
    Successful request count: 470
    Avg request latency: 519608163228 usec (overhead 519608155497 usec + queue 111 usec + compute input 3902 usec + compute infer 3645 usec + compute output 73 usec)

Inferences/Second vs. Client Average Batch Latency
Concurrency: 1, throughput: 78.4 infer/sec, latency 12744 usec

As the overhead time mentioned by @dselivanov in #1989 , the input dims were dims: [3, 2048, 2048] and the overhead time was only 67.62230263157895us, but mine was very much longer.

I also used the python to test. 
### python  infer time
preprocess time: 0.0029985904693603516
infer time: 0.9250223636627197
preprocess time: 0.0029108524322509766
infer time: 1.1599602699279785
preprocess time: 0.0030019283294677734
infer time: 0.957021951675415
preprocess time: 0.0029993057250976562
infer time: 1.0499982833862305
preprocess time: 0.003998994827270508
infer time: 0.618964433670044

Hope someone to help, Thanks very much~

**Comment by dselivanov** [comment]
In my case problem resolved with explicit setting output placeholder with`binary_data=True`:
```py
InferRequestedOutput('output', binary_data=True)
```

**Comment by satchelwu** [comment]
> In my case problem resolved with explicit setting output placeholder with`binary_data=True`:
> 
> ```python
> InferRequestedOutput('output', binary_data=True)
> ```

Thanks!
I tryied, but it did not work. the InferRequestedOutput's param of  binary_data is true by default.  The performance test result above was evaluated by the c++ perf_client. The single model inference time in Tesla T4 was about 11ms, but in triton, the overhead time is so long. It is so confused.

**Comment by deadeyegoodwin** [comment]
Note that based on your throughput numbers the overhead time cannot be correct. Something is going wrong in its collection. What version of perf_client / perf_analyzer are you using?

**Comment by satchelwu** [comment]
> Note that based on your throughput numbers the overhead time cannot be correct. Something is going wrong in its collection. What version of perf_client / perf_analyzer are you using?

"git log" shows that 
### git log
commit 3159e73d9173c5c67119d3a9f16b7cb46d668e8c (HEAD, tag: v2.3.0, origin/r20.09)
Author: dzier <dzier@nvidia.com>
Date:   Thu Sep 17 14:20:25 2020 -0700

    Update README and versions for 20.09 release

**Comment by satchelwu** [comment]
I used cmake to build triton. I wonder if there are some specific cmake options.  I only used -DTRITON_ENABLE_GPU -DTRITON_ENABEL_TENSORRT -DCMAKE_PREFIX_INSTALL

**Comment by tanmayv25** [comment]
@satchelwu Can you verify the following:
1. Running the server in verbose mode, do you see any errors getting reported?
2. Invoking the perf_client with `-v -v` flags will print super verbose output with all the json responses received from the server. Can you search for the message `inference_stats` in the logs and share its contents? This will help in narrowing down the issue to being at the client-side vs server side. My guess is something is going wrong with the server.

Can you share the `test_model` so that I may try to reproduce the issue?

**Comment by tanmayv25** [comment]
With your settings, your actual `Avg request latency` should be less than `Avg HTTP time` which is **12744us**. This entails a maximum overhead to be **5013usec**.

**Comment by satchelwu** [comment]
> @satchelwu Can you verify the following:
> 
> 1. Running the server in verbose mode, do you see any errors getting reported?
> 2. Invoking the perf_client with `-v -v` flags will print super verbose output with all the json responses received from the server. Can you search for the message `inference_stats` in the logs and share its contents? This will help in narrowing down the issue to being at the client-side vs server side stat collection. My guess is something is going wrong with the server which might be messing up with the statistics collection.
> 
> Can you share the `test_model` so that I may try to reproduce the issue?

@tanmayv25 Thanks very much!
1.I followed your advice to start the server with verbose mode by "**--log-info=true --log-warning=true  --log-verbose=5 --log-error=true**" and no error was found.
#server start cmd output
I1022 01:27:52.385810 2272 metrics.cc:184] found 2 GPUs supporting NVML metrics
I1022 01:27:52.394645 2272 metrics.cc:193]   GPU 0: Tesla T4
I1022 01:27:52.400307 2272 metrics.cc:193]   GPU 1: Tesla T4
I1022 01:27:52.400514 2272 server.cc:120] Initializing Triton Inference Server
I1022 01:27:52.400534 2272 server.cc:121]   id: 'triton'
I1022 01:27:52.400547 2272 server.cc:122]   version: '2.3.0'
I1022 01:27:52.400561 2272 server.cc:128]   extensions:  classification sequence model_repository schedule_policy model_configuration system_shared_memory cuda_shared_memory binary_tensor_data statistics
I1022 01:27:52.596492 2272 pinned_memory_manager.cc:195] Pinned memory pool is created at '0x7f8910000000' with size 268435456
I1022 01:27:52.597159 2272 cuda_memory_manager.cc:98] CUDA memory pool is created on device 0 with size 67108864
I1022 01:27:52.597167 2272 cuda_memory_manager.cc:98] CUDA memory pool is created on device 1 with size 67108864
I1022 01:27:52.674679 2272 plan_backend_factory.cc:48] Create PlanBackendFactory
I1022 01:27:52.674703 2272 plan_backend_factory.cc:55] Registering TensorRT Plugins
I1022 01:27:52.674777 2272 logging.cc:52] Registered plugin creator - ::GridAnchor_TRT version 1
I1022 01:27:52.674789 2272 logging.cc:52] Registered plugin creator - ::NMS_TRT version 1
I1022 01:27:52.674798 2272 logging.cc:52] Registered plugin creator - ::Reorg_TRT version 1
I1022 01:27:52.674810 2272 logging.cc:52] Registered plugin creator - ::Region_TRT version 1
I1022 01:27:52.674820 2272 logging.cc:52] Registered plugin creator - ::Clip_TRT version 1
I1022 01:27:52.674828 2272 logging.cc:52] Registered plugin creator - ::LReLU_TRT version 1
I1022 01:27:52.674838 2272 logging.cc:52] Registered plugin creator - ::PriorBox_TRT version 1
I1022 01:27:52.674847 2272 logging.cc:52] Registered plugin creator - ::Normalize_TRT version 1
I1022 01:27:52.674856 2272 logging.cc:52] Registered plugin creator - ::RPROI_TRT version 1
I1022 01:27:52.674866 2272 logging.cc:52] Registered plugin creator - ::BatchedNMS_TRT version 1
I1022 01:27:52.674875 2272 logging.cc:52] Registered plugin creator - ::FlattenConcat_TRT version 1
I1022 01:27:52.674884 2272 logging.cc:52] Registered plugin creator - ::CropAndResize version 1
I1022 01:27:52.674893 2272 logging.cc:52] Registered plugin creator - ::DetectionLayer_TRT version 1
I1022 01:27:52.674902 2272 logging.cc:52] Registered plugin creator - ::Proposal version 1
I1022 01:27:52.674911 2272 logging.cc:52] Registered plugin creator - ::ProposalLayer_TRT version 1
I1022 01:27:52.674920 2272 logging.cc:52] Registered plugin creator - ::PyramidROIAlign_TRT version 1
I1022 01:27:52.674931 2272 logging.cc:52] Registered plugin creator - ::ResizeNearest_TRT version 1
I1022 01:27:52.674938 2272 logging.cc:52] Registered plugin creator - ::Split version 1
I1022 01:27:52.674946 2272 logging.cc:52] Registered plugin creator - ::SpecialSlice_TRT version 1
I1022 01:27:52.674955 2272 logging.cc:52] Registered plugin creator - ::InstanceNormalization_TRT version 1
I1022 01:27:52.674965 2272 ensemble_backend_factory.cc:47] Create EnsembleBackendFactory
I1022 01:27:52.676209 2272 model_repository_manager.cc:595] AsyncLoad() 'test_model'
I1022 01:27:52.676230 2272 model_repository_manager.cc:657] TriggerNextAction() 'test_model' version 1: 1
I1022 01:27:52.676237 2272 model_repository_manager.cc:695] Load() 'test_model' version 1
I1022 01:27:52.676258 2272 model_repository_manager.cc:714] loading: test_model:1
I1022 01:27:52.676390 2272 model_repository_manager.cc:767] CreateInferenceBackend() 'test_model' version 1
I1022 01:27:54.775063 2272 logging.cc:52] Deserialize required 1195015 microseconds.
I1022 01:27:54.775107 2272 plan_backend.cc:333] Creating instance test_model_0_0_gpu1 on GPU 1 (7.5) using model.plan
I1022 01:27:54.778659 2272 plan_backend.cc:509] Detected data as execution binding for test_model
I1022 01:27:54.778675 2272 plan_backend.cc:509] Detected prob as execution binding for test_model
I1022 01:27:54.779023 2272 plan_backend.cc:643] Created instance test_model_0_0_gpu1 on GPU 1 with stream priority 0
I1022 01:27:54.779203 2272 plan_backend.cc:333] Creating instance test_model_0_1_gpu1 on GPU 1 (7.5) using model.plan
I1022 01:27:54.782379 2272 plan_backend.cc:509] Detected data as execution binding for test_model
I1022 01:27:54.782391 2272 plan_backend.cc:509] Detected prob as execution binding for test_model
I1022 01:27:54.782687 2272 plan_backend.cc:643] Created instance test_model_0_1_gpu1 on GPU 1 with stream priority 0
I1022 01:27:54.782864 2272 plan_backend.cc:333] Creating instance test_model_0_2_gpu1 on GPU 1 (7.5) using model.plan
I1022 01:27:54.785931 2272 plan_backend.cc:509] Detected data as execution binding for test_model
I1022 01:27:54.785943 2272 plan_backend.cc:509] Detected prob as execution binding for test_model
I1022 01:27:54.786247 2272 plan_backend.cc:643] Created instance test_model_0_2_gpu1 on GPU 1 with stream priority 0
I1022 01:27:54.786423 2272 plan_backend.cc:333] Creating instance test_model_0_3_gpu1 on GPU 1 (7.5) using model.plan
I1022 01:27:54.789409 2272 plan_backend.cc:509] Detected data as execution binding for test_model
I1022 01:27:54.789422 2272 plan_backend.cc:509] Detected prob as execution binding for test_model
I1022 01:27:54.789704 2272 plan_backend.cc:643] Created instance test_model_0_3_gpu1 on GPU 1 with stream priority 0
I1022 01:27:54.789861 2272 plan_backend.cc:333] Creating instance test_model_0_4_gpu1 on GPU 1 (7.5) using model.plan
I1022 01:27:54.793111 2272 plan_backend.cc:509] Detected data as execution binding for test_model
I1022 01:27:54.793124 2272 plan_backend.cc:509] Detected prob as execution binding for test_model
I1022 01:27:54.793414 2272 plan_backend.cc:643] Created instance test_model_0_4_gpu1 on GPU 1 with stream priority 0
I1022 01:27:54.793574 2272 plan_backend.cc:333] Creating instance test_model_0_5_gpu1 on GPU 1 (7.5) using model.plan
I1022 01:27:54.796673 2272 plan_backend.cc:509] Detected data as execution binding for test_model
I1022 01:27:54.796685 2272 plan_backend.cc:509] Detected prob as execution binding for test_model
I1022 01:27:54.796982 2272 plan_backend.cc:643] Created instance test_model_0_5_gpu1 on GPU 1 with stream priority 0
I1022 01:27:54.797143 2272 plan_backend.cc:333] Creating instance test_model_0_6_gpu1 on GPU 1 (7.5) using model.plan
I1022 01:27:54.800310 2272 plan_backend.cc:509] Detected data as execution binding for test_model
I1022 01:27:54.800324 2272 plan_backend.cc:509] Detected prob as execution binding for test_model
I1022 01:27:54.800599 2272 plan_backend.cc:643] Created instance test_model_0_6_gpu1 on GPU 1 with stream priority 0
I1022 01:27:54.800760 2272 plan_backend.cc:333] Creating instance test_model_0_7_gpu1 on GPU 1 (7.5) using model.plan
I1022 01:27:54.805007 2272 plan_backend.cc:509] Detected data as execution binding for test_model
I1022 01:27:54.805019 2272 plan_backend.cc:509] Detected prob as execution binding for test_model
I1022 01:27:54.805290 2272 plan_backend.cc:643] Created instance test_model_0_7_gpu1 on GPU 1 with stream priority 0
I1022 01:27:54.805446 2272 dynamic_batch_scheduler.cc:225] Starting dynamic-batch scheduler thread 0 at nice 5...
I1022 01:27:54.805546 2272 plan_backend.cc:358] plan backend for test_model
name=test_model
contexts:
  name=test_model_0_0_gpu1, gpu=1, max_batch_size=8
  bindings:
    0: max possible byte_size=23592960, buffer=0x7f8890000000 ]
    1: max possible byte_size=192032, buffer=0x7f88b7e4d400 ]
  name=test_model_0_1_gpu1, gpu=1, max_batch_size=8
  bindings:
    0: max possible byte_size=23592960, buffer=0x7f888e000000 ]
    1: max possible byte_size=192032, buffer=0x7f88b7ec4e00 ]
  name=test_model_0_2_gpu1, gpu=1, max_batch_size=8
  bindings:
    0: max possible byte_size=23592960, buffer=0x7f881c000000 ]
    1: max possible byte_size=192032, buffer=0x7f88b7f3c800 ]
  name=test_model_0_3_gpu1, gpu=1, max_batch_size=8
  bindings:
    0: max possible byte_size=23592960, buffer=0x7f8808000000 ]
    1: max possible byte_size=192032, buffer=0x7f88b7fb4200 ]
  name=test_model_0_4_gpu1, gpu=1, max_batch_size=8
  bindings:
    0: max possible byte_size=23592960, buffer=0x7f8806800000 ]
    1: max possible byte_size=192032, buffer=0x7f8809a40000 ]
  name=test_model_0_5_gpu1, gpu=1, max_batch_size=8
  bindings:
    0: max possible byte_size=23592960, buffer=0x7f87f2800000 ]
    1: max possible byte_size=192032, buffer=0x7f8809aaf000 ]
  name=test_model_0_6_gpu1, gpu=1, max_batch_size=8
  bindings:
    0: max possible byte_size=23592960, buffer=0x7f87de800000 ]
    1: max possible byte_size=192032, buffer=0x7f8809b1e000 ]
  name=test_model_0_7_gpu1, gpu=1, max_batch_size=8
  bindings:
    0: max possible byte_size=23592960, buffer=0x7f87ca800000 ]
    1: max possible byte_size=192032, buffer=0x7f8809b92a00 ]

I1022 01:27:54.818100 2272 model_repository_manager.cc:896] successfully loaded 'test_model' version 1
I1022 01:27:54.818112 2272 model_repository_manager.cc:657] TriggerNextAction() 'test_model' version 1: 0
I1022 01:27:54.818117 2272 model_repository_manager.cc:672] no next action, trigger OnComplete()
I1022 01:27:54.818233 2272 model_repository_manager.cc:492] VersionStates() 'test_model'
I1022 01:27:54.819896 2272 grpc_server.cc:224] Ready for RPC 'ServerLive', 0
I1022 01:27:54.819937 2272 grpc_server.cc:224] Ready for RPC 'ServerReady', 0
I1022 01:27:54.819951 2272 grpc_server.cc:224] Ready for RPC 'ModelReady', 0
I1022 01:27:54.819966 2272 grpc_server.cc:224] Ready for RPC 'ServerMetadata', 0
I1022 01:27:54.819982 2272 grpc_server.cc:224] Ready for RPC 'ModelMetadata', 0
I1022 01:27:54.820000 2272 grpc_server.cc:224] Ready for RPC 'ModelConfig', 0
I1022 01:27:54.820018 2272 grpc_server.cc:224] Ready for RPC 'ModelStatistics', 0
I1022 01:27:54.820033 2272 grpc_server.cc:224] Ready for RPC 'SystemSharedMemoryStatus', 0
I1022 01:27:54.820051 2272 grpc_server.cc:224] Ready for RPC 'SystemSharedMemoryRegister', 0
I1022 01:27:54.820069 2272 grpc_server.cc:224] Ready for RPC 'SystemSharedMemoryUnregister', 0
I1022 01:27:54.820084 2272 grpc_server.cc:224] Ready for RPC 'CudaSharedMemoryStatus', 0
I1022 01:27:54.820106 2272 grpc_server.cc:224] Ready for RPC 'CudaSharedMemoryRegister', 0
I1022 01:27:54.820121 2272 grpc_server.cc:224] Ready for RPC 'CudaSharedMemoryUnregister', 0
I1022 01:27:54.820142 2272 grpc_server.cc:224] Ready for RPC 'RepositoryIndex', 0
I1022 01:27:54.820160 2272 grpc_server.cc:224] Ready for RPC 'RepositoryModelLoad', 0
I1022 01:27:54.820174 2272 grpc_server.cc:224] Ready for RPC 'RepositoryModelUnload', 0
I1022 01:27:54.820208 2272 grpc_server.cc:357] Thread started for CommonHandler
I1022 01:27:54.820413 2272 grpc_server.cc:3000] New request handler for ModelInferHandler, 1
I1022 01:27:54.820468 2272 grpc_server.cc:2078] Thread started for ModelInferHandler
I1022 01:27:54.820602 2272 grpc_server.cc:3345] New request handler for ModelStreamInferHandler, 3
I1022 01:27:54.820654 2272 grpc_server.cc:2078] Thread started for ModelStreamInferHandler
I1022 01:27:54.820669 2272 grpc_server.cc:3897] Started GRPCInferenceService at 0.0.0.0:8001
I1022 01:27:54.821289 2272 http_server.cc:2705] Started HTTPService at 0.0.0.0:8000
I1022 01:27:54.863168 2272 http_server.cc:2724] Started Metrics Service at 0.0.0.0:8002

2. It seemed that no output was printed after running perf_client with **-v** .  
 ./perf_client -m test_model -u 192.168.54.181:8000 -v
*** Measurement Settings ***
  Batch size: 1
  Measurement window: 5000 msec
  Using synchronous calls for inference
  Stabilizing using average latency

Request concurrency: 1
  Pass [1] throughput: 8.4 infer/sec. Avg latency: 129138 usec (std 74107 usec)
  Pass [2] throughput: 9.6 infer/sec. Avg latency: 105795 usec (std 6947 usec)
  Pass [3] throughput: 9.2 infer/sec. Avg latency: 108136 usec (std 7425 usec)
  Pass [4] throughput: 8.8 infer/sec. Avg latency: 114044 usec (std 10739 usec)
  Client:
    Request count: 44
    Throughput: 8.8 infer/sec
    Avg latency: 114044 usec (standard deviation 10739 usec)
    p50 latency: 113176 usec
    p90 latency: 123916 usec
    p95 latency: 126632 usec
    p99 latency: 155897 usec
    Avg HTTP time: 112408 usec (send 1093 usec + response wait 110880 usec + receive 435 usec)
  Server:
    Inference count: 53
    Execution count: 53
    Successful request count: 53
    Avg request latency: 776538458634 usec (overhead 776538442544 usec + queue 124 usec + compute input 3725 usec + compute infer 12169 usec + compute output 72 usec)

Inferences/Second vs. Client Average Batch Latency
Concurrency: 1, throughput: 8.8 infer/sec, latency 114044 usec


I suddenly realized that **1000ms / Avg HTTP time == infer/sec** inspired by the Avg HTTP time you mentioned. your guess about server collecting statistics might be right.

@tanmayv25 Thanks very much again!
Thanks all!

**Comment by tanmayv25** [comment]
@satchelwu With that figured out, we still need to identify why perf_client is reporting such large server side latency. I am sorry for not being very clear. Can you run the following and share your observations? Note that -v is specified twice for super-verbosity.

```
./perf_client -m test_model -u 192.168.54.181:8000 -v -v 2>&1 >> client.log
cat client.log | grep "model_stats"
```
Just share the output of the second command.

**Comment by satchelwu** [comment]
> @satchelwu With that figured out, we still need to identify why perf_client is reporting such large server side latency. I am sorry for not being very clear. Can you run the following and share your observations? Note that -v is specified twice for super-verbosity.
> 
> ```
> ./perf_client -m test_model -u 192.168.54.181:8000 -v -v 2>&1 >> client.log
> cat client.log | grep "model_stats"
> ```
> 
> Just share the output of the second command.

The server is not available right now. I'll try it once available.

**Comment by satchelwu** [comment]
@tanmayv25 So sorry for waiting. the server is ready now, and the output is below.
./perf_client -m test_model -u 192.168.54.181:8000 -v -v 2>&1 >> client.log
### cat client.log | grep "model_stats"
{"model_stats":[{"name":"test_model","version":"1","last_inference":0,"inference_count":0,"execution_count":0,"inference_stats":{"success":{"count":0,"ns":0},"fail":{"count":0,"ns":0},"queue":{"count":0,"ns":0},"compute_input":{"count":0,"ns":0},"compute_infer":{"count":0,"ns":0},"compute_output":{"count":0,"ns":0}},"batch_stats":[]}]}
inference response: {"model_stats":[{"name":"test_model","version":"1","last_inference":1603682378753,"inference_count":18,"execution_count":18,"inference_stats":{"success":{"count":18,"ns":81063907530424},"fail":{"count":0,"ns":0},"queue":{"count":18,"ns":2615789},"compute_input":{"count":18,"ns":857581106},"compute_infer":{"count":18,"ns":206504973},"compute_output":{"count":18,"ns":1246042}},"batch_stats":[{"batch_size":1,"compute_input":{"count":18,"ns":857581106},"compute_infer":{"count":18,"ns":206504973},"compute_output":{"count":18,"ns":1246042}}]}]}{"model_name":"test_model","model_version":"1","outputs":[{"name":"prob","datatype":"FP32","shape":[1,6001,1,1],"parameters":{"binary_data_size":24004}}]}
{"model_stats":[{"name":"test_model","version":"1","last_inference":1603682378753,"inference_count":18,"execution_count":18,"inference_stats":{"success":{"count":18,"ns":81063907530424},"fail":{"count":0,"ns":0},"queue":{"count":18,"ns":2615789},"compute_input":{"count":18,"ns":857581106},"compute_infer":{"count":18,"ns":206504973},"compute_output":{"count":18,"ns":1246042}},"batch_stats":[{"batch_size":1,"compute_input":{"count":18,"ns":857581106},"compute_infer":{"count":18,"ns":206504973},"compute_output":{"count":18,"ns":1246042}}]}]}
{"model_stats":[{"name":"test_model","version":"1","last_inference":1603682384785,"inference_count":50,"execution_count":50,"inference_stats":{"success":{"count":50,"ns":225341376007243},"fail":{"count":0,"ns":0},"queue":{"count":50,"ns":6445144},"compute_input":{"count":50,"ns":976942568},"compute_infer":{"count":50,"ns":595327999},"compute_output":{"count":50,"ns":3464469}},"batch_stats":[{"batch_size":1,"compute_input":{"count":50,"ns":976942568},"compute_infer":{"count":50,"ns":595327999},"compute_output":{"count":50,"ns":3464469}}]}]}
{"model_stats":[{"name":"test_model","version":"1","last_inference":1603682384785,"inference_count":50,"execution_count":50,"inference_stats":{"success":{"count":50,"ns":225341376007243},"fail":{"count":0,"ns":0},"queue":{"count":50,"ns":6445144},"compute_input":{"count":50,"ns":976942568},"compute_infer":{"count":50,"ns":595327999},"compute_output":{"count":50,"ns":3464469}},"batch_stats":[{"batch_size":1,"compute_input":{"count":50,"ns":976942568},"compute_infer":{"count":50,"ns":595327999},"compute_output":{"count":50,"ns":3464469}}]}]}
{"model_stats":[{"name":"test_model","version":"1","last_inference":1603682390849,"inference_count":93,"execution_count":93,"inference_stats":{"success":{"count":93,"ns":419480128024861},"fail":{"count":0,"ns":0},"queue":{"count":93,"ns":11558140},"compute_input":{"count":93,"ns":1133669384},"compute_infer":{"count":93,"ns":1121544651},"compute_output":{"count":93,"ns":6427571}},"batch_stats":[{"batch_size":1,"compute_input":{"count":93,"ns":1133669384},"compute_infer":{"count":93,"ns":1121544651},"compute_output":{"count":93,"ns":6427571}}]}]}
{"model_stats":[{"name":"test_model","version":"1","last_inference":1603682390957,"inference_count":94,"execution_count":94,"inference_stats":{"success":{"count":94,"ns":423997895856029},"fail":{"count":0,"ns":0},"queue":{"count":94,"ns":11678505},"compute_input":{"count":94,"ns":1137263627},"compute_infer":{"count":94,"ns":1133766678},"compute_output":{"count":94,"ns":6491403}},"batch_stats":[{"batch_size":1,"compute_input":{"count":94,"ns":1137263627},"compute_infer":{"count":94,"ns":1133766678},"compute_output":{"count":94,"ns":6491403}}]}]}
{"model_stats":[{"name":"test_model","version":"1","last_inference":1603682397083,"inference_count":139,"execution_count":139,"inference_stats":{"success":{"count":139,"ns":627432499399681},"fail":{"count":0,"ns":0},"queue":{"count":139,"ns":17048102},"compute_input":{"count":139,"ns":1305414399},"compute_infer":{"count":139,"ns":1682409338},"compute_output":{"count":139,"ns":9627781}},"batch_stats":[{"batch_size":1,"compute_input":{"count":139,"ns":1305414399},"compute_infer":{"count":139,"ns":1682409338},"compute_output":{"count":139,"ns":9627781}}]}]}
{"model_stats":[{"name":"test_model","version":"1","last_inference":1603682397083,"inference_count":139,"execution_count":139,"inference_stats":{"success":{"count":139,"ns":627432499399681},"fail":{"count":0,"ns":0},"queue":{"count":139,"ns":17048102},"compute_input":{"count":139,"ns":1305414399},"compute_infer":{"count":139,"ns":1682409338},"compute_output":{"count":139,"ns":9627781}},"batch_stats":[{"batch_size":1,"compute_input":{"count":139,"ns":1305414399},"compute_infer":{"count":139,"ns":1682409338},"compute_output":{"count":139,"ns":9627781}}]}]}
{"model_stats":[{"name":"test_model","version":"1","last_inference":1603682403256,"inference_count":184,"execution_count":184,"inference_stats":{"success":{"count":184,"ns":831157853034258},"fail":{"count":0,"ns":0},"queue":{"count":184,"ns":21643542},"compute_input":{"count":184,"ns":1469319586},"compute_infer":{"count":184,"ns":2233709017},"compute_output":{"count":184,"ns":12724726}},"batch_stats":[{"batch_size":1,"compute_input":{"count":184,"ns":1469319586},"compute_infer":{"count":184,"ns":2233709017},"compute_output":{"count":184,"ns":12724726}}]}]}

**Comment by tanmayv25** [comment]
**"success":{"count":139,"ns":627432499399681}**
The ludicrously large value of this stat indicates there is a bug in tritonserver when capturing request_start and request end timestamps. I have been unable to reproduce this bug, and I don't see anything explicitly wrong with the implementation that catches my eye.

@satchelwu Can you share the test_model so that I can run some debugging on my end?

**Comment by satchelwu** [comment]
> **"success":{"count":139,"ns":627432499399681}**
> The ludicrously large value of this stat indicates there is a bug in tritonserver when capturing request_start and request end timestamps. I have been unable to reproduce this bug, and I don't see anything explicitly wrong with the implementation that catches my eye.
> 
> @satchelwu Can you share the test_model so that I can run some debugging on my end?

here is my model, and custom plugin
[](https://pan.baidu.com/s/171VIVu417Xy_QfNssPyhqw) 
password: 83p3

**Comment by tanmayv25** [comment]
I have fixed a bug in the stat collection and perf_client report generation. It will be most likely be available in our 20.11 release.  As a workaround, you can include the following cmake configuration flag:
`-DTRITON_ENABLE_TRACING=ON`

**Comment by satchelwu** [comment]
> I have fixed a bug in the stat collection and perf_client report generation. It will be most likely be available in our 20.11 release. As a workaround, you can include the following cmake configuration flag while building the server for now:
> `-DTRITON_ENABLE_TRACING=ON`

Thanks very much!
